mlflow/tracking/client.py

```diff
@@ -650,38 +690,57 @@ def load_prompt(
                 )
             name, version = self.parse_prompt_uri(name_or_uri)
         else:
+            if version is None:
+                raise MlflowException(
+                    "Version must be specified when loading a prompt by name. "
+                    "Use a prompt URI (e.g., 'prompts:/name/version') or provide the version "
+                    "parameter.",
+                    INVALID_PARAMETER_VALUE,
+                )
             name = name_or_uri
 
         registry_client = self._get_registry_client()
         try:
-            mv = (
-                registry_client.get_latest_versions(name, stages=ALL_STAGES)[0]
-                if version is None
-                else registry_client.get_model_version(name, version)
-            )
+            # Use get_prompt_version for specific version/alias
+            return registry_client.get_prompt_version(name, version)
+
         except MlflowException as exc:
             if allow_missing and exc.error_code == "RESOURCE_DOES_NOT_EXIST":
                 return None
             raise
 
-        # Fetch the prompt-level tags from the registered model
-        prompt_tags = registry_client.get_registered_model(name)._tags
-
-        return PromptVersion.from_model_version(mv, prompt_tags=prompt_tags)
-
+    @deprecated(
+        since="3.0",
+        alternative="delete_prompt_version",
+    )
     @experimental
     @require_prompt_registry
     @translate_prompt_exception
     def delete_prompt(self, name: str, version: int):
```

Comment:
Let's remove this. We can remove methods in 3.0; if we don't remove, we'll have to keep it around until 4.0 :/